### PR TITLE
feat(server): domain step in pandas engine

### PR DESCRIPTION
**Pull request by davinov**
Starting point of any pipeline :)

I also added a very light playground server, that we later can connect to the front-end playground, as replacement of the mongo-based one.

### CODE REVIEW

File: server/setup.cfg [hunk 1]
```diff
@@ -9,3 +9,9 @@ ignore = E203, E266, E501, W503
 include_trailing_comma = True
 line_length = 100
 multi_line_output = 3
+
+[mypy]
+plugins = pydantic.mypy
+follow_imports = silent
+ignore_missing_imports = True
+allow_redefinition = True
```
Comment (fspot) [inline server/setup.cfg:17]:
```suggestion
allow_redefinition = True
check_untyped_defs = True
```
pratique pour avoir du typecheck même sans signature (par exemple sur les tests)

File: server/weaverbird/pipeline_executor.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    """
+    The main class of the module.
+
+    Its domain retriever will be use to retrieve data needed from `domains` steps.
+    """
+
+    def __init__(self, domain_retriever: Callable[[str], pd.DataFrame]):
+        self.retrieve_domain = domain_retriever
+
+    def execute_pipeline(self, pipeline: List[dict]):
```
Comment (fspot) [inline server/weaverbird/pipeline_executor.py:18]:
```suggestion
    def execute_pipeline(self, pipeline: List[dict]) -> pd.DataFrame:
```

File: server/tests/test_pipeline_executor.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+DOMAINS = {'domain_a': pd.read_csv(path.join(path.dirname(__file__), 'fixtures/domain_a.csv'))}
+
+
+@pytest.fixture
+def pipeline_executor():
+    return PipelineExecutor(domain_retriever=lambda name: DOMAINS[name])
+
+
+def test_extract_domain(pipeline_executor):
```
Comment (fspot) [inline server/tests/test_pipeline_executor.py:17]:
```suggestion
def test_extract_domain(pipeline_executor: PipelineExecutor):
```

**Review by fspot — APPROVED** [review]
Quelque détails d'annotations de type mais sinon lgtm :+1:

File: server/setup.cfg (hunk 1 above)
Comment (davinov) [reply server/setup.cfg:17]: https://github.com/ToucanToco/weaverbird/pull/670/commits/a1c81bfa3012c195f55c66eaaa0f1e9b551c91ad

File: server/weaverbird/pipeline_executor.py (hunk 2 above)
Comment (davinov) [reply server/weaverbird/pipeline_executor.py]: https://github.com/ToucanToco/weaverbird/pull/670/commits/debc2cdda36b131656859d331e251c2b64349f69

File: server/tests/test_pipeline_executor.py (hunk 3 above)
Comment (davinov) [reply server/tests/test_pipeline_executor.py]: https://github.com/ToucanToco/weaverbird/pull/670/commits/debc2cdda36b131656859d331e251c2b64349f69

**Pull request merged**